Source/WebCore/rendering/RenderBox.cpp

```diff
@@ -1394,6 +1394,84 @@ void RenderBox::clearOverridingLogicalWidthLength()
         gOverridingLogicalWidthLengthMap->remove(this);
 }
 
+FlowRelativeDirection RenderBox::physicalToFlowRelativeDirectionMapping(PhysicalDirection direction) const
+{
+    auto isHorizontalWritingMode = style().isHorizontalWritingMode();
+    auto isLeftToRightDirection = style().isLeftToRightDirection();
+
+    if (isHorizontalWritingMode == containingBlock()->style().isHorizontalWritingMode()) {
+        switch (direction) {
+        case PhysicalDirection::Top:
+            return isHorizontalWritingMode ? FlowRelativeDirection::BlockStart : (isLeftToRightDirection ? FlowRelativeDirection::InlineStart : FlowRelativeDirection::InlineEnd);
```

Comment:
At some point this function needs to be revisited and figure out what this mapping really means since if you try to map directions on *this, it's actually never based on *this style, but the context *this is in (parent, containing block etc depending on the layout system) for example, an inline-block with trimmed-margin can have "direction: rtl" but the direction applies to its _content_ and not to the inline-block as an inline level box (i.e. it does not make the inline level box (initiated by the inline-block) suddenly "right direction" on the line).